<launch>
  <arg name="config" default="planar"/>
  <arg name="port" default="js4"/>
  <node pkg="interactive_marker_twist_server" type="marker_server" name="twist_marker_server" args="_robot_name:=apollo">
    <remap from="twist_marker_server/cmd_vel" to="/cmd_vel"/>
    <rosparam command="load" file="$(find interactive_marker_twist_server)/config/$(arg config).yaml"/>
  </node>
  <node pkg="apollo_control" type="twist_limiter.py" name="limiter"/>
  <rosparam command ="load" file="$(find apollo_control)/config/teleop.yaml"/>
  <node name="joy_node" pkg="joy" type="joy_node">
    <param name="dev" type="string" value="/dev/input/$(arg port)"/>
    <param name="autorepeat_rate" type="double" value="50"/>
  </node>
  <node name="manual_controller" pkg="apollo_control" type="manual_teleop.py" output="screen" args="cmd_vel:=manual_teleop/cmd_vel"/>
  <node pkg="twist_mux" type="twist_mux" name="twist_mux">
    <rosparam command="load" file="$(find apollo_control)/config/twist_mux.yaml"/>
    <remap from="cmd_vel_out" to="cmd_vel"/>
  </node>
</launch>
